File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/items/menu/NotifyOnFlipViewAnimator.java
package com.lasthopesoftware.bluewater.client.browsing.library.items.menu;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ViewAnimator;

/**
 * Created by david on 10/28/15.
 */
public class NotifyOnFlipViewAnimator extends ViewAnimator {

    private OnViewChangedListener onViewChangedListener;

    public NotifyOnFlipViewAnimator(Context context) {
        super(context);
    }

    public NotifyOnFlipViewAnimator(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public void setDisplayedChild(final int whichChild) {
        super.setDisplayedChild(whichChild);

        onViewChangedListener.onViewChanged(this);
    }

    public void setViewChangedListener(OnViewChangedListener onViewChangedListener) {
        this.onViewChangedListener = onViewChangedListener;
    }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/menu/NotifyOnFlipViewAnimator.kt
package com.lasthopesoftware.bluewater.client.browsing.items.menu

import android.content.Context
import android.util.AttributeSet
import android.widget.ViewAnimator

/**
 * Created by david on 10/28/15.
 */
class NotifyOnFlipViewAnimator : ViewAnimator {
    private var onViewChangedListener: OnViewChangedListener? = null

    constructor(context: Context) : super(context)
	constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

	override fun setDisplayedChild(whichChild: Int) {
        super.setDisplayedChild(whichChild)
        onViewChangedListener?.onViewChanged(this)
    }

    fun setViewChangedListener(onViewChangedListener: OnViewChangedListener?) {
        this.onViewChangedListener = onViewChangedListener
    }
}
